### PR TITLE
add support for PidsLimit #734

### PR DESCRIPTION
**Pull request by msekunda**
Changes for #734

This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/764)

### CODE REVIEW

**Comment by cirocosta** [comment]
Any news on this?

File: src/test/java/com/github/dockerjava/core/command/CreateContainerCmdImplTest.java [hunk 1]
```diff
@@ -753,4 +755,26 @@ public void createContainerWithShmSize() throws DockerException {
...
+        CreateContainerResponse container = dockerClient.createContainerCmd(BUSYBOX_IMAGE)
+                                                        .withHostConfig(hostConfig).withCmd("true").exec();
+
+        LOG.info("Created container {}", container.toString());
+
+        assertThat(container.getId(), not(isEmptyString()));
+
+        InspectContainerResponse inspectContainerResponse = dockerClient.inspectContainerCmd(container.getId()).exec();
+
+        assertEquals(inspectContainerResponse.getHostConfig().getPidsLimit(), hostConfig.getPidsLimit());
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/core/command/CreateContainerCmdImplTest.java]: please use hamcrest matcher.

File: src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java [hunk 2]
```diff
@@ -611,6 +611,13 @@ public CreateContainerCmd withCpuShares(Integer cpuShares) {
     }
 
     @Override
+    public CreateContainerCmd withPidsLimit(Long pidsLimit) {
+        checkNotNull(pidsLimit, "pidsLimit was not specified");
+        hostConfig.withPidsLimit(pidsLimit);
+        return this;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java]: the third PR... You can configure hostConfig directly without proxy calls from create object.

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java]: So even in testcase you create and configure hostConfig directly without calling proxy methods.

**Comment by KostyaSha** [comment]
And we end that it already supported and PR has only style formatting?

**Comment by KostyaSha** [comment]
Sorry, now looks good :)

**Comment by msekunda** [comment]
Should be fine, sorry for formatting problem :).

**Pull request merged**